test/Datadog.Trace.Tests/PlatformHelpers/AzureAppServicesMetadataTests.cs

```diff
@@ -0,0 +1,145 @@
+        public void ResourceId_IsNull_WhenAnyRequirementsMissing(string subscriptionId, string deploymentId, string siteResourceGroup)
+        {
+            // plan resource group actually doesn't matter for the resource id we build
+            var vars = GetMockVariables(subscriptionId, deploymentId, "some-resource-group", siteResourceGroup);
+            var metadata = new AzureAppServices(vars);
+            Assert.Null(metadata.ResourceId);
+        }
+
+        [Fact]
+        public void PopulatesOnlyRootSpans()
+        {
+            var vars = GetMockVariables(SubscriptionId, DeploymentId, PlanResourceGroup, SiteResourceGroup);
+            AzureAppServices.Metadata = new AzureAppServices(vars);
+            var tracer = new Tracer();
+            var rootSpans = new List<Span>();
+            var nonRootSpans = new List<Span>();
+            var iterations = 5;
+            var remaining = iterations;
+
+            while (remaining-- > 0)
+            {
+                using (var rootScope = tracer.StartActive("root"))
+                {
+                    rootSpans.Add(rootScope.Span);
+
+                    using (var nestedScope = tracer.StartActive("nest-a"))
+                    {
+                        nonRootSpans.Add(nestedScope.Span);
+                    }
+
+                    using (var nestedScope = tracer.StartActive("nest-b"))
+                    {
+                        nonRootSpans.Add(nestedScope.Span);
+
+                        using (var doublyNestedScope = tracer.StartActive("nest-b-1"))
+                        {
+                            nonRootSpans.Add(doublyNestedScope.Span);
+                        }
+                    }
+                }
+            }
+
+            Assert.Equal(expected: iterations, actual: rootSpans.Count);
+            Assert.NotEmpty(nonRootSpans);
+
+            var rootSpansMissingExpectedTag =
+                rootSpans.Where(s => s.GetTag(Tags.AzureAppServicesResourceId) != ExpectedResourceId).ToList();
+
+            var nonRootSpansWithTag =
+                nonRootSpans.Where(s => s.GetTag(Tags.AzureAppServicesResourceId) == ExpectedResourceId);
+
+            var newLine = Environment.NewLine;
+            var detailedMessage =
+                string.Join(
+                    Environment.NewLine,
+                    rootSpansMissingExpectedTag.Select(
+                        r => $"Expected {ExpectedResourceId} {newLine}but received {r.GetTag(Tags.AzureAppServicesResourceId) ?? "NULL"} {newLine}{newLine}({r}) {newLine}"));
+
+            var envVarValues = string.Join(", ", EnvVars.Select(e => $"{e}: {Environment.GetEnvironmentVariable(e)}"));
+
+            Assert.True(!rootSpansMissingExpectedTag.Any(), $"All root spans should have the resource id: {newLine}{envVarValues}{newLine}{detailedMessage}");
+            Assert.True(!nonRootSpansWithTag.Any(), "No non root spans should have the resource id.");
+
+            ClearVariables();
```

Comment:
nit: looks like we don't need this anymore (and any test changing env vars makes me nervous)

```suggestion
```